src/log-hook.ts

```diff
@@ -269,4 +277,36 @@ export function hookLoggerToMongoLogWriter(
       'Missing ID token in IdP response'
     );
   });
+
+  emitter.on('mongodb-oidc-plugin:outbound-http-request', (ev) => {
+    log.debug?.(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_023),
+      `${contextPrefix}-oidc`,
+      'Outbound HTTP request',
+      { url: redactUrl(ev.url) }
+    );
+  });
+
+  emitter.on('mongodb-oidc-plugin:inbound-http-request', (ev) => {
+    log.debug?.(
+      'OIDC-PLUGIN',
+      mongoLogId(1_002_000_024),
+      `${contextPrefix}-oidc`,
+      'Inbound HTTP request',
+      { url: redactUrl(ev.url) }
+    );
+  });
+}
+
+function redactUrl(url: string): string {
+  let parsed: URL;
+  try {
+    parsed = new URL(url);
+  } catch {
+    return '<Invalid URL>';
```

Comment:
NIT: Would be nice to log.debug? somehow the wrong URL or at least return the invalid url in case there is an issue.

```suggestion
    return `<Invalid URL: '${url}'>`;
```

Reply:
@kmruiz Well ... I've gone back and forth on this, the point of this function is to redact URLs so that sensitive parameters like the authorization code don't end up being logged (and consequently stored to disk), so I wanted to err on the side of caution. You're also obviously right that that makes for bad diagnostics, not having the URL available at all :) I'd lean a bit towards leaving it like this, this condition Should Never Happen™ anyway because we should always be working with valid URLs, but let me know if you feel differently

Reply:
I merged this for now, but happy to adjust in a follow-up if you have more thoughts on this :)